syft/core/workers/base.py

```diff
@@ -607,10 +607,11 @@ def get_worker(self, id_or_worker):
             if id_or_worker in self._known_workers:
                 return self._known_workers[id_or_worker]
             else:
-                logging.warning(
-                    "Worker", self.id, "couldnt recognize worker", id_or_worker
+                raise TypeError(
```

Comment:
Should this be a type error?

Reply:
To be honest, I'm not particularly convinced of this as well. In some places a "`Exception` is raised, but I don't think this is the best approach. Maybe raise a `RuntimeWarning`  (https://docs.python.org/2/library/exceptions.html#exceptions.RuntimeWarning) ?

Reply:
Pure Exceptions should never be raised as we could always be more specific. A runtime warning makes sense to me

Reply:
Done.